### PR TITLE
User demo

### PR DESCRIPTION
**Pull request by nehal-mtaj**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 class ApplicationController < ActionController::Base
+  def after_sign_in_path
+   root_path
+ end
   protect_from_forgery with: :exception
 end
```
Comment (ManojParmar-MTAJ) [inline app/controllers/application_controller.rb]: write all custom methods after this line not before this

File: app/controllers/homes_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class HomesController < ApplicationController
+    before_action :authenticate_user!
```
Comment (ManojParmar-MTAJ) [inline app/controllers/homes_controller.rb]: fix indentation and add one blank line before writing custom methods

File: app/models/user_demo.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class UserDemo < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+end
```
Comment (ManojParmar-MTAJ) [inline app/models/user_demo.rb]: i think no need for this file now...

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <div class="col-xs-8">
+          <div class="checkbox icheck">
+            <label>
+             <!-- <%= f.check_box :remember_me %> Remember Me -->
+            </label>
+          </div>
+        </div>
+        <div class="col-xs-4">
+          <%= f.submit "Sign In", class: 'btn btn-primary btn-block btn-flat' %>
+                  </div>
```
Comment (ManojParmar-MTAJ) [inline app/views/devise/sessions/new.html.erb]: fix indentation

File: app/views/devise/shared/_links.html.erb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (ManojParmar-MTAJ) [inline app/views/devise/shared/_links.html.erb]: remove first blank line

File: app/views/devise/shared/_links.html.erb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+<%- if devise_mapping.lockable? && resource_class.unlock_strategy_enabled?(:email) && controller_name != 'unlocks' %>
+<%= link_to "Didn't receive unlock instructions?", new_unlock_path(resource_name) %><br />
+<% end -%>
+
+<%- if devise_mapping.omniauthable? %>
+<%- resource_class.omniauth_providers.each do |provider| %>
+<%= link_to "Sign in with #{OmniAuth::Utils.camelize(provider)}", omniauth_authorize_path(resource_name, provider) %><br />
+<% end -%>
+<% end -%>
```
Comment (ManojParmar-MTAJ) [inline app/views/devise/shared/_links.html.erb]: fix indentation code add tab between if..end

File: config/routes.rb [hunk 7]
```diff
@@ -1,5 +1,17 @@
 Rails.application.routes.draw do
-  root 'homes#index'
+  root to: 'homes#index'
```
Comment (ManojParmar-MTAJ) [inline config/routes.rb]: theme is integrated now so change root path to products_path and remove routes and code related to home controller

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -247,7 +247,7 @@
         <a href="#" class="btn btn-default btn-flat">Profile</a>
       </div>
       <div class="pull-right">
-        <a href="#" class="btn btn-default btn-flat">Sign out</a>
+        <%= link_to 'Sign Out', destroy_user_session_path ,:method => :delete %>
```
Comment (RaviMariya-MTAJ) [inline app/views/layouts/_header.html.erb]: use new rails syntex instead of `:method => :delete` use `method: :delete`

**Pull request merged**